src/app/main/component/user/components/profile/profile-cards/profile-cards.component.ts

```diff
@@ -101,8 +110,8 @@ export class ProfileCardsComponent implements OnInit, OnDestroy {
       )
       .subscribe((fact: FactOfTheDay | null) => {
         if (fact) {
-          this.factOfTheDay = fact;
           this.localStorageService.saveFactToLocalStorage(fact, currentTime, this.factKey);
+          this.factOfTheDay = this.getFactContentByLanguage(this.factKey);
```

Comment:
DRY: similar logic in `fetchAndSaveHabitFact` and `fetchAndSaveRandomFact`
SRP: these methods not only get data but also save it
Can we use one approach to obtain data?
something like this:
```
  private fetchAndSaveFactOfTheDay(currentTime: number, key: string, url: string): void {
    this.profileService
      .getFactOfTheDay(url)
      .pipe(
        catchError(() => of(null)),
        takeUntil(this.destroy$)
      )
      .subscribe((fact: FactOfTheDay) => {
        if (fact) {
          this.localStorageService.saveFactToLocalStorage(fact, currentTime, key);
        }
      });
  }
  
    updateFacts(currentTime: number, hasHabits: boolean): void {
    this.fetchAndSaveFactOfTheDay(currentTime, this.factKey, '/by-tags');
    this.factOfTheDay = this.getFactContentByLanguage(this.factKey);

    if (hasHabits) {
      this.fetchAndSaveFactOfTheDay(currentTime, this.habitFactKey, '');
      this.habitFactOfTheDay = this.getFactContentByLanguage(this.habitFactKey);
    }
  }
  ```
  and we can update service by removing getFactsOfTheDayByTags
  ```
    getRandomFactOfTheDay(url: string): Observable<FactOfTheDay> {
    const currentLang = this.languageService.getCurrentLanguage();
    return this.http.get<FactOfTheDay>(`${mainLink}fact-of-the-day/random${url}?lang=${currentLang}`);
  }
  ```